File: Memento/Golang/Editor.go
package main

import (
	"fmt"
)

type Editor struct {
	state string
}

func (editor *Editor) setState(state string) {
	editor.state = state
}

func (editor *Editor) getState() string {
	return editor.state
}

func (editor *Editor) save() *WorkspaceState {
	return NewWorkspaceState(editor.getState())
}

func (editor *Editor) restore(workspaceState *WorkspaceState)  {
	if workspaceState != nil {
		editor.setState(workspaceState.getState())
	}
}

func (editor *Editor) print() {
	fmt.Println(editor.getState() + "\n\n")
}

File: Memento/Javascript/Editor.js
import WorkspaceState from "./WorkspaceState.js";

class Editor {
  constructor() {
    this.__state = null;
  }

  set_state(state) {
    this.__state = state;
  }

  save() {
    if (this.__state) {
      return new WorkspaceState(this.__state);
    } else {
      console.log("No state to save");
    }
  }

  restore(state) {
    this.__state = state.get_state();
  }

  print() {
    console.log(this.__state);
  }
}

export default Editor;